conan/tools/premake/toolchain.py

```diff
@@ -0,0 +1,245 @@
+import os
+import textwrap
+from pathlib import Path
+
+from jinja2 import Environment
+
+from conan.tools.build.cross_building import cross_building
+from conan.tools.files import save
+from conan.tools.microsoft.visual import VCVars
+from conan.tools.premake.premakedeps import PREMAKE_ROOT_FILE
+
+_jinja_env = Environment(trim_blocks=True, lstrip_blocks=True)
```

Comment:
No need for a global here, this can be instantiated in the necessary places, not that important to reuse